pkg/apis/pingcap/v1alpha1/component_spec.go

```diff
@@ -80,11 +80,16 @@ type ComponentAccessor interface {
 }
 
 func (tc *TidbCluster) AllComponentSpec() []ComponentAccessor {
-	components := []ComponentAccessor{}
+	var components []ComponentAccessor
 	components = append(components, tc.BaseDiscoverySpec())
 	if tc.Spec.PD != nil {
 		components = append(components, tc.BasePDSpec())
 	}
+	if tc.Spec.PDMS != nil {
+		for _, service := range tc.Spec.PDMS {
+			components = append(components, tc.BasePDMSSpec(service))
+		}
+	}
```

Comment:
Can the line of `if tc.Spec.PDMS != nil {` be removed directly? just `for ... range` on a nil slice should be OK.